src/Scrutor/Decoration/DecoratorInstanceFactory.cs

```diff
@@ -0,0 +1,22 @@
+﻿using Microsoft.Extensions.DependencyInjection;
+using System;
+
+namespace Scrutor.Decoration
+{
+    internal static class DecoratorInstanceFactory
```

Comment:
I get Enterprise Java vibes from all the InstanceFactoryProvider-esque naming 😅 

Proposal:

- Rename `Decoration` to `Decorator`.
- Drop `DecorationFactory` and move the methods to `Decorator` (as static methods), i.e. `Decorator.Create`.
- Convert `IDecorationStrategy` to an abstract `DecoratorStrategy` class.
- Drop `DecoratorInstanceFactory` and move the methods to protected `DecoratorStrategy` methods.